randovania/games/prime2/layout/beam_configuration.py

```diff
@@ -1,31 +1,26 @@
 import dataclasses
 from typing import Iterator
 
+from open_prime_rando.dol_patching.echoes.beam_configuration import BeamAmmoConfiguration
 from randovania.bitpacking.bitpacking import BitPackDataclass
 from randovania.bitpacking.json_dataclass import JsonDataclass
 from randovania.bitpacking.type_enforcement import DataclassPostInitTypeCheck
 
 
 @dataclasses.dataclass(frozen=True)
-class BeamAmmoConfiguration(BitPackDataclass, JsonDataclass, DataclassPostInitTypeCheck):
-    item_index: int = dataclasses.field(metadata={"min": 0, "max": 108})
-    ammo_a: int = dataclasses.field(metadata={"min": -1, "max": 108})
-    ammo_b: int = dataclasses.field(metadata={"min": -1, "max": 108})
-    uncharged_cost: int = dataclasses.field(metadata={"min": 0, "max": 250, "if_different": 1})
-    charged_cost: int = dataclasses.field(metadata={"min": 0, "max": 250, "if_different": 5})
-    combo_missile_cost: int = dataclasses.field(metadata={"min": 1, "max": 250, "if_different": 5})
-    combo_ammo_cost: int = dataclasses.field(metadata={"min": 0, "max": 250, "if_different": 30})
+class RdvBeamAmmoConfiguration(BeamAmmoConfiguration, BitPackDataclass, JsonDataclass, DataclassPostInitTypeCheck):
```

Comment:
do we intend to move these all out into a separate library(ies)? if we do I assume we'll just roll all these mixins into the base BeamAmmoConfiguration in opr

Reply:
Maybe at some point, but I'm skipping doing that extra package right now.